smol-backend/src/Smol/Backend/IR/FromExpr/Expr.hs

```diff
@@ -133,35 +136,31 @@ addVar ::
 addVar ident expr =
   modify (\s -> s {fesVars = fesVars s <> M.singleton ident expr})
 
-{-
-lookupVar ::
-  (MonadState (FromExprState ann) m) =>
-  IRIdentifier ->
-  m IRExpr
-lookupVar ident = do
-  maybeExpr <- gets (M.lookup ident . vars)
-  case maybeExpr of
-    Just a -> pure a
-    _ -> error $ "could not find var " <> show ident
--}
-
--- in which we turn our higher level language into the middle level silly
--- language
--- we should create the main function, which always prints its result
--- a side effect will be that it adds IRModuleParts to the state, these are all
--- combined
-irFromExpr ::
-  (Show ann) =>
-  Map (Identity TypeName) (DataType Identity ann) ->
-  IdentityExpr (Type Identity ann) ->
-  IRModule
-irFromExpr dataTypes expr =
-  IRModule $
-    [ IRExternDef $ getPrinter (getExprAnnotation expr),
-      IRExternDef irStringConcat,
-      IRExternDef irStringEquals -- we should dynamically include these once we get a lot of stdlib helpers
-    ]
-      <> modulePartsFromExpr dataTypes expr
+-- | turn a Smol module into an IR one
+-- no considerations for name collisions etc
+irFromModule :: (Show ann) => Module ResolvedDep (Type ResolvedDep ann) -> IRModule
```

Comment:
This should go in a new file tbh, let's get it in and then move things around a bit.